File: src/main/java/net/moltendorf/Bukkit/LikeDynomite/LikeDynomite.java
package net.moltendorf.Bukkit.LikeDynomite;

import org.bukkit.Server;
import org.bukkit.plugin.PluginManager;
import org.bukkit.plugin.java.JavaPlugin;

/**
 *
 * @author moltendorf
 */
public class LikeDynomite extends JavaPlugin {
	protected static LikeDynomite instance;

	// Variable data.
	protected Configuration configuration = null;

	@Override
	public void onEnable() {
		instance = this;

		// Construct new configuration.
		configuration = new Configuration();

		// Are we enabled?
		if (!configuration.global.enabled) {
			return;
		}

		// Get server.
		final Server server = getServer();

		// Get plugin manager.
		final PluginManager manager = server.getPluginManager();

		// Register our event listeners.
		manager.registerEvents(new Listeners(), this);
	}

	@Override
	public void onDisable() {
		instance = null;
	}
}

File: src/main/kotlin/net/moltendorf/bukkit/likedynomite/LikeDynomite.kt
package net.moltendorf.bukkit.likedynomite

import org.bukkit.plugin.java.JavaPlugin

/**

 * @author moltendorf
 */
class LikeDynomite : JavaPlugin() {
  lateinit var settings: Settings

  override fun onEnable() {
    instance = this

    // Construct new configuration.
    settings = Settings()

    // Are we enabled?
    enabled = settings.enabled

    if (enabled) {
      // Get plugin manager.
      val manager = server.pluginManager

      // Register listeners.
      manager.registerEvents(Listeners(), this)
    }
  }

  override fun onDisable() {
    enabled = false
  }

  companion object {
    var enabled = false
      private set

    lateinit var instance: LikeDynomite
      private set
  }
}
